dual_net.py

```diff
@@ -209,9 +209,11 @@ def model_fn(features, labels, mode, params=None):
             logits=logits, labels=tf.stop_gradient(labels['pi_tensor'])))
     value_cost = tf.reduce_mean(
         tf.square(value_output - labels['value_tensor']))
-    l2_cost = FLAGS.l2_strength * tf.add_n([
-        tf.nn.l2_loss(v)
-        for v in tf.trainable_variables() if not 'bias' in v.name])
+    reg_vars = [v for v in tf.trainable_variables()
+                if not '/bias:' in v.name and not '/beta:' in v.name]
```

Comment:
i'm having trouble parsing this boolean logic. So the variables you want to regularize are:
- All variables
- except for bias variables
- but actually, regularize bias variables if they're part of batchnorm

I think this is slightly less complicated, although it's still a bit hard to read.
`v for v in trainable_variables() if 'batch' in v.name or not 'bias' in v.name`

I think the right solution is to abstract out a function `should_regularize(v)` and write unit tests for it to sanity check that we're doing what we think we're doing..

Reply:
You can see the list of variables in https://github.com/tensorflow/minigo/pull/294#issuecomment-400506907

currently `tf.trainable_variables` is:
batch_normalization_<X>/beta:0
batch_normalization_<X>/gamma:0
conv2d_<X>/bias:0
conv2d_<X>/kernel:0
dense_<X>/bias:0
dense_<X>/kernel:0

But we exclude the conv2d and dense bias. I thought it made sense that we should also exclude batch_normalization beta (which I thought was essentially a bias)